src/deno_dir.rs

```diff
@@ -156,7 +168,19 @@ impl DenoDir {
         None => Ok(()),
       }?;
       deno_fs::write_file(&p, &source, 0o666)?;
-      deno_fs::write_file(&mt, content_type.as_bytes(), 0o666)?;
+      // Remove possibly existing stale .mime file
+      // may not exist. DON'T unwrap
+      let _ = std::fs::remove_file(&media_type_filename);
+      // Create .mime file only when content type different from extension
+      let resolved_content_type = map_content_type(&p, Some(&content_type));
+      let dot_index = filename.rfind(".").unwrap_or(0);
+      let ext = filename.chars().skip(dot_index).collect::<String>();
```

Comment:
Is there not some dedicated std function for extracting the extension ?

Reply:
See if you can use this
https://doc.rust-lang.org/std/path/struct.Path.html#method.extension